core/src/main/java/com/sequenceiq/distrox/v1/distrox/StackOperations.java

```diff
@@ -145,139 +146,140 @@ public StackV4Response get(@NotNull StackAccessDto stackAccessDto, Long workspac
-        LOGGER.info("Query Stack (view) successfully finished with crn {}", stackAccessDto.getCrn());
+        ResourceAccessDto.validate(resourceAccessDto);
+        StackApiView stackApiView = stackApiViewService.retrieveStackByCrnAndType(resourceAccessDto.getCrn(), stackType);
+        LOGGER.info("Query Stack (view) successfully finished with crn {}", resourceAccessDto.getCrn());
         StackViewV4Response stackViewV4Response = converterUtil.convert(stackApiView, StackViewV4Response.class);
         LOGGER.info("Adding environment name to the response.");
         environmentServiceDecorator.prepareEnvironment(stackViewV4Response);
         return stackViewV4Response;
     }
 
-    public void deleteInstance(@NotNull StackAccessDto stackAccessDto, Long workspaceId, boolean forced, String instanceId) {
-        if (isNotEmpty(stackAccessDto.getName())) {
-            stackCommonService.deleteInstanceByNameInWorkspace(stackAccessDto.getName(), workspaceId, instanceId, forced);
+    public void deleteInstance(@NotNull ResourceAccessDto resourceAccessDto, Long workspaceId, boolean forced, String instanceId) {
+        if (isNotEmpty(resourceAccessDto.getName())) {
+            stackCommonService.deleteInstanceByNameInWorkspace(resourceAccessDto.getName(), workspaceId, instanceId, forced);
         } else {
-            stackCommonService.deleteInstanceByCrnInWorkspace(stackAccessDto.getCrn(), workspaceId, instanceId, forced);
+            stackCommonService.deleteInstanceByCrnInWorkspace(resourceAccessDto.getCrn(), workspaceId, instanceId, forced);
         }
     }
 
-    public void deleteInstances(StackAccessDto stackAccessDto, Long workspaceId, List<String> instanceIds, boolean forced) {
-        if (isNotEmpty(stackAccessDto.getName())) {
-            stackCommonService.deleteMultipleInstancesByNameInWorkspace(stackAccessDto.getName(), workspaceId, instanceIds, forced);
+    public void deleteInstances(ResourceAccessDto resourceAccessDto, Long workspaceId, List<String> instanceIds, boolean forced) {
+        if (isNotEmpty(resourceAccessDto.getName())) {
+            stackCommonService.deleteMultipleInstancesByNameInWorkspace(resourceAccessDto.getName(), workspaceId, instanceIds, forced);
         } else {
-            stackCommonService.deleteMultipleInstancesByCrnInWorkspace(stackAccessDto.getCrn(), workspaceId, instanceIds, forced);
+            stackCommonService.deleteMultipleInstancesByCrnInWorkspace(resourceAccessDto.getCrn(), workspaceId, instanceIds, forced);
         }
     }
 
-    public void sync(@NotNull StackAccessDto stackAccessDto, Long workspaceId) {
-        if (isNotEmpty(stackAccessDto.getName())) {
-            stackCommonService.syncInWorkspace(stackAccessDto.getName(), null, workspaceId);
+    public void sync(ResourceAccessDto resourceAccessDto, Long workspaceId) {
+        if (isNotEmpty(resourceAccessDto.getName())) {
```

Comment:
I would make just one call, as syncInWorkspace handles the either name or crn case.

Reply:
also refactor topic, not related to this specific fix